src/main.rs

```diff
@@ -949,6 +1037,77 @@ async fn main() {
         );
     }
 
+    tokio::spawn(async {
+        let local_port = match env::var("LAVA_GITLAB_RUNNER_LOCAL_PORT") {
+            Ok(val) => val
+                .parse()
+                .expect("failed to parse LAVA_GITLAB_RUNNER_LOCAL_PORT as a port number"),
+            Err(_) => {
+                warn!("No LAVA_GITLAB_RUNNER_LOCAL_PORT set, will listen on ephemeral IP.");
+                0u16
+            }
+        };
+
+        let listener = std::net::TcpListener::bind(std::net::SocketAddr::new(
+            IpAddr::V4(Ipv4Addr::UNSPECIFIED),
+            local_port,
+        ))
+        .expect("failed to bind listener");
+
+        let routable_host = match env::var("LAVA_GITLAB_RUNNER_ROUTABLE_HOST") {
```

Comment:
This should really just be one argument/environment variable  for the base URL rather then specifying a seperate hostname and port so one can also do `https://my-upload-host:2456/badger/snakes` if  this endpoint is not mounted on the root